src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -1083,16 +1094,14 @@ void RealmCoordinator::run_async_notifiers()
         //  - Notifier C has a source version of 5
         // Notifier A wants the changes from versions 2-latest, B wants 7-latest,
         // and C wants 5-latest. We achieve this by starting at version 2 and
-        // attaching A, then advancing to version 5 (letting A gather changes
+        // attaching A, then advancing to version 5 (letting A gather cemplacehanges
         // from 2-5). We then attach C and advance to 7, then attach B and advance
         // to the latest.
-        std::sort(new_notifiers.begin(), new_notifiers.end(), compare_notifier_versions);
-        new_notifier_transaction = m_db->start_read(new_notifiers.front()->version());
+        new_notifier_transaction = m_db->start_read(version);
```

Comment:
How come sorting is no longer needed?

Reply:
Sorting is done when constructing `IncrementalChangeInfo` just below. Line 929